src/RestWrite.js

```diff
@@ -248,7 +249,21 @@ RestWrite.prototype.handleAuthDataValidation = function(authData) {
       throw new Parse.Error(Parse.Error.UNSUPPORTED_SERVICE,
         'This authentication method is unsupported.');
     }
-    return validateAuthData(authData[provider]);
+    return validateAuthData(authData[provider]).then(() => {
+      if (this.response) {
+      // it is a login call
+        runBeforeLoginHandler({
```

Comment:
As mentioned previously, the object passed to the beforeLogin hook should match the ones provided in other hooks like beforeSave http://parseplatform.org/Parse-SDK-JS/api/v1.11.0/Parse.Cloud.html#.TriggerRequest

The req.object property should be an instance of Parse.User, all the key types should match their proper types (createdAt/updatedAt as dates), authProvider don’t make sense on req.object but may be left on the top level.

We also expect all other keys like installationId, headers etc.. to be set

Reply:
Hi @flovilmart, sorry for the incomplete PR, I misunderstood your message. I am fixing it.
Thanks,